packages/material-ui/src/Portal/Portal.js

```diff
@@ -17,65 +17,60 @@ function getOwnerDocument(element) {
  * Portals provide a first-class way to render children into a DOM node
  * that exists outside the DOM hierarchy of the parent component.
  */
-class Portal extends React.Component {
-  componentDidMount() {
-    this.setMountNode(this.props.container);
+function Portal(props) {
+  const {children, disablePortal, onRendered, container} = props;
+  const [, forceUpdate] = React.useState(0);
+  const mountNode = React.useRef();
+  const renderedTimer = React.useRef();
+
+  const setMountNode = React.useCallback(() => {
+    if (disablePortal) {
+      mountNode.current = ReactDOM.findDOMNode(this).parentElement;
+
+      return;
+    }
+    mountNode.current = getContainer(container, getOwnerDocument(this).body);
+  },                                     [container, disablePortal]);
+
+  const getMountNode = () => mountNode.current;
```

Comment:
Do we need a function, why not using `mountNode.current` directly?

Reply:
done, Please review.